src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -97,11 +103,11 @@ void IDkmClrResultProvider.GetItems(DkmEvaluationResultEnumContext enumContext,
                     return;
                 }
 
-                completionRoutine(GetItems(enumContext.InspectionContext, dataItem.EvalResultDataItem, startIndex, count));
+                completionRoutine(GetItems(dataItem.EvalResultDataItem, workList, startIndex, count, enumContext.InspectionContext));
             }
-            catch (Exception e) when (ExpressionEvaluatorFatalError.CrashIfFailFastEnabled(e))
+            catch (Exception e) when (ExpressionEvaluatorFatalError.ReportNonFatalException(e, DkmComponentManager.ReportCurrentNonFatalException))
             {
-                throw ExceptionUtilities.Unreachable;
+                completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e));
```

Comment:
Thanks!  That explains a lot.  I've updated all our continuations to try/catch (and removed the try/catch where the Dispatcher should be handling it).